src/signed_data.rs

```diff
@@ -352,84 +352,124 @@ pub static RSA_PSS_2048_8192_SHA384_LEGACY_KEY: SignatureAlgorithm = SignatureAl
 /// Requires the `alloc` feature.
 #[cfg(feature = "alloc")]
 pub static RSA_PSS_2048_8192_SHA512_LEGACY_KEY: SignatureAlgorithm = SignatureAlgorithm {
-    public_key_alg_id: RSA_ENCRYPTION,
-    signature_alg_id: RSA_PSS_SHA512,
+    public_key_alg_id: alg_id::RSA_ENCRYPTION,
+    signature_alg_id: alg_id::RSA_PSS_SHA512,
     verification_alg: &signature::RSA_PSS_2048_8192_SHA512,
 };
 
 /// ED25519 signatures according to RFC 8410
 pub static ED25519: SignatureAlgorithm = SignatureAlgorithm {
-    public_key_alg_id: ED_25519,
-    signature_alg_id: ED_25519,
+    public_key_alg_id: alg_id::ED_25519,
+    signature_alg_id: alg_id::ED_25519,
     verification_alg: &signature::ED25519,
 };
 
-struct AlgorithmIdentifier {
-    asn1_id_value: untrusted::Input<'static>,
-}
-
-impl AlgorithmIdentifier {
-    fn matches_algorithm_id_value(&self, encoded: untrusted::Input) -> bool {
-        encoded == self.asn1_id_value
+/// Encodings of the PKIX AlgorithmIdentifier type:
+///
+/// ```text
+/// AlgorithmIdentifier  ::=  SEQUENCE  {
+///     algorithm               OBJECT IDENTIFIER,
+///     parameters              ANY DEFINED BY algorithm OPTIONAL  }
+///                                -- contains a value of the type
+///                                -- registered for use with the
+///                                -- algorithm object identifier value
+/// ```
+///
+/// The outer sequence encoding is not included, so this is an OID encoding
+/// for `algorithm` plus the `parameters` value.
+pub mod alg_id {
```

Comment:
What does using an inline module buy us here?

Reply:
for me its about separating the pile of constants (that are not useful to most users of the library, exported for implementers of the trait) and the useful bits with similar names. for example, without this module we'd have `webpki::ED25519` and `webpki::ED_25519` (the latter is the algorithm identifier, and afaict has the underscore on main because they live in the same module -- just about to delete the underscore now i've noticed it.)

Reply:
(could put these in a separate file if that is preferred)

Reply:
Makes sense, maybe add a comment to make this more obvious?